BlueprintUI/Sources/Layout/Builder/StackElement+Builder.swift

```diff
@@ -0,0 +1,115 @@
+import UIKit
+
+/// `StackChild` is a wrapper which allows an element to define its StackLayout traits and Keys.
+/// This struct is particularly useful when working with `@StackElementBuilder`.
+/// By default, elements will default to a nil key and the default `StackLayout.Traits` initializer.
+/// `@StackElementBuilder` will check every child to see if it can be type cast to a `StackChild`
+/// and then pull of the given traits and key and then apply those to the stack
+public struct StackChild: ProxyElement {
+    public var traits: StackLayout.Traits
+    public var key: AnyHashable?
+    private let wrappedElement: Element
+
+    public init(
+        wrappedElement: Element,
+        traits: StackLayout.Traits = .init(),
+        key: AnyHashable? = nil
+    ) {
+        self.wrappedElement = wrappedElement
+        self.traits = traits
+        self.key = key
+    }
+    
+    public init(
+        wrappedElement: Element,
+        growPriority: CGFloat = 1.0,
```

Comment:
Nit: We generally push people towards the existing `addFixed` and `addFlexible` methods on the Stack – should we use that paradigm here too? Eg thinking of maybe an enum we could use or something, like:

```
    public init(
        wrappedElement: Element,
        sizing: Sizing,
        ...
    )

```

Where `Sizing` is like:

```
struct Sizing {
   static let fixed : Self = .init(0, 0)
   static let flexible : Self = .init(1, 1)

   var grow : CGFloat
   var shrink : CGFloat
}
```

Reply:
Nice! I like this approach.